larsim/EventGenerator/RadioGen_module.cc

```diff
@@ -160,32 +273,93 @@ namespace {
   constexpr double m_neutron = 0.9395654133; // mass of a neutron in GeV
 }
 
+
 namespace evgen{
 
   //____________________________________________________________________________
   RadioGen::RadioGen(fhicl::ParameterSet const& pset)
     : EDProducer{pset}
-    , fNuclide {pset.get< std::vector<std::string>>("Nuclide")}
-    , fMaterial{pset.get< std::vector<std::string>>("Material")}
-    , fBq{pset.get< std::vector<double> >("BqPercc")}
-    , fT0{pset.get< std::vector<double> >("T0")}
-    , fT1{pset.get< std::vector<double> >("T1")}
-    , fX0{pset.get< std::vector<double> >("X0")}
-    , fY0{pset.get< std::vector<double> >("Y0")}
-    , fZ0{pset.get< std::vector<double> >("Z0")}
-    , fX1{pset.get< std::vector<double> >("X1")}
-    , fY1{pset.get< std::vector<double> >("Y1")}
-    , fZ1{pset.get< std::vector<double> >("Z1")}
+    , fX0{pset.get< std::vector<double> >("X0", {})}
+    , fY0{pset.get< std::vector<double> >("Y0", {})}
+    , fZ0{pset.get< std::vector<double> >("Z0", {})}
+    , fX1{pset.get< std::vector<double> >("X1", {})}
+    , fY1{pset.get< std::vector<double> >("Y1", {})}
+    , fZ1{pset.get< std::vector<double> >("Z1", {})}
     , fIsFirstSignalSpecial{pset.get< bool >("IsFirstSignalSpecial", false)}
     // create a default random engine; obtain the random seed from NuRandomService,
     // unless overridden in configuration with key "Seed"
     , fEngine(art::ServiceHandle<rndm::NuRandomService>{}->createEngine(*this, pset, "Seed"))
   {
     produces< std::vector<simb::MCTruth> >();
     produces< sumdata::RunData, art::InRun >();
-
+    
+    
+    auto const nuclide = pset.get< std::vector<std::string>>("Nuclide");
+    auto const material = pset.get< std::vector<std::string>>("Material");
+    auto const Bq = pset.get< std::vector<double> >("BqPercc");
+    auto t0 = pset.get< std::vector<double> >("T0", {});
+    auto t1 = pset.get< std::vector<double> >("T1", {});
+    
+    if (fT0.empty() || fT1.empty()) { // better be both empty...
+      assert(fT0.empty() && fT1.empty());
+      auto const times = defaulttimewindow();
+      t0.push_back(times.first);
+      t1.push_back(times.second);
+    }
```

Comment:
A curious and problematic formulation.  Presumably, you intend something like:
```.cpp
if (size(t0) != size(t1)) {
  throw ...;
}

if (empty(t0)) {
  auto const [start, stop] = defaulttimewindow();
  t0.push_back(start);
  t1.push_back(stop);
}
```

Reply:
Fair enough.
The choice was intentional but not well motivated (and probably an abuse of `assert()`).
Pushed a change along the suggested lines.